pdf/pdfannotation.cpp

```diff
@@ -89,6 +97,9 @@ void PDFAnnotation::attachOnce(PDFDocument *document, int page)
 
 void PDFAnnotation::attach(PDFDocument *document, PDFSelection *selection)
```

Comment:
This seems to guard against most things but what if `d->m_document == nullptr && document == nullptr`? I think this would then call `addAnnotation()` on `nullptr`.

Reply:
I'll add null check for the document parameter too.